### DISCUSSION TITLE
Restart app doesn't respect processArguments

### DISCUSSION THREAD

**Discussion by PureWeen** [appium/appium-dotnet-driver#858] [Q&A] [answered]
I noticed this issue from a few years back. I'm having a similar issue now that everything has been moved to activateApp

https://github.com/appium/appium/issues/16852

1) Whenever I reset the app the Env/processArgs don't get passed into the set app
2) is there a way to influence these in the way that's described on the issue using the newer APIs? Do I need to just use "Execute" and call launchApp myself?

**Comment by KazuCocoa** [comment, marked as answer]
> Do I need to just use "Execute" and call launchApp myself?

Yes, https://appium.github.io/appium-xcuitest-driver/latest/reference/execute-methods/#mobile-launchapp is intended to be used to start an app with processArguments/environment variables.

**Reply by PureWeen** [reply]
Wasn't launchapp removed and deprecated from this library? 
https://github.com/appium/dotnet-client/pull/766

So, do I just do it via my own call to "Execute" ?

Is there a friendly method for this path? Is launch app not recommended though since it was deprecated?

**Reply by Dor-bl** [reply]
It will be useful if you can raise a PR for a friendly method.

**Reply by PureWeen** [reply]
Sounds good!! I was just making sure I wasn't missing something obvious here. 
So as far as implementation on the API would the path here be to add an "ActivateApp" overload on the iOSDriver ?

Since LaunchApp was deprecated? 

Or would it be something along the lines of this https://github.com/appium/dotnet-client/pull/813 

Add LaunchApp to the iOSDriver specifically that maps to 
https://appium.github.io/appium-xcuitest-driver/latest/reference/execute-methods/#mobile-launchapp

**Reply by Dor-bl** [reply]
It depends if launchApp execute command is supported on other drivers as well, I need to confirm it.
But if it's only for iOS we will need a dedicated method just for iOS.

**Reply by Dor-bl** [reply]
@KazuCocoa launchApp command is also supported on uiautomator 2 driver?
@mykola-mokhnach fyi

**Reply by KazuCocoa** [reply]
No, the equivalent one in uia2 is startActivity. https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#mobile-startactivity

**Reply by Dor-bl** [reply]
@PureWeen,
We got confirmation, so you can go ahead and implement the command only for the iOS driver.

**Reply by PureWeen** [reply]
alright, I'll get it figured out in my app here then take that learning forward :-)

**Reply by Dor-bl** [reply]
@PureWeen, I know it's been a while, but any news from your side?
